build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/upgrades/UpgradedProperties.java

```diff
@@ -104,6 +104,29 @@ private static boolean isOldBooleanGetterOfUpgradedProperty(JApiMethod jApiMetho
         return isOldMethod(jApiMethod, upgradedMethods, BOOLEAN_GETTER_REGEX) && jApiMethod.getReturnType().getOldReturnType().equals("boolean");
     }
 
+    private static boolean isKotlinBooleanSourceCompatibilityMethod(JApiMethod jApiMethod, Map<AccessorKey, UpgradedProperty> currentMethods) {
```

Comment:
❓  it would be nice to cover this with a test. I assume that is intended via `getIsFailOnError`, but I think it should actually fail since it's a java source, not Kotlin.

Reply:
Yes, this `getIsFailOnError()` tests that. 
Actually these methods are not written in Kotlin, but in Java. They can be added there just to use these methods in Kotlin is source compatible

Reply:
See example:
https://github.com/gradle/gradle/blob/a904b80dcd068cb06a873807aad4cdd9287875fe/platforms/jvm/language-jvm/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java#L104-L118